File: C#/Template/WordCVReportGeneration.cs
using Template.Models;

namespace Template;

public sealed class WordCVReportGeneration : CvReportGeneration
{
    /// <summary>
    /// Image Scanning implementation
    /// </summary>
    /// <param name="cvPath"><inheritdoc/></param>
    /// <returns><inheritdoc/></returns>
    protected override Data ExtractData(string cvPath)
    {
        Console.WriteLine("Extracting data from word...");
        return new Data();
    }
}

File: Python/template/WordCVReportGeneration.py
from CVReportGeneration import CVReportGeneration
from io import FileIO as File
from Data import Data

class WordCVReportGeneration(CVReportGeneration):
    def __init__(self):
        super().__init__()

    def _extract_data(self, file: File) -> Data:
        print(f"Extracting data from a Word file...")
        return None